pilot/pkg/model/extensions.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package model
+
+import (
+	"strings"
+	"time"
+
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_extensions_filters_http_wasm_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/wasm/v3"
+	envoy_extensions_wasm_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/wasm/v3"
+	"github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	extensions "istio.io/api/extensions/v1alpha1"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/util/gogo"
+)
+
+const (
+	defaultRuntime = "envoy.wasm.runtime.v8"
+)
+
+type WasmPluginWrapper struct {
+	extensions.WasmPlugin
+
+	Name      string
+	Namespace string
+
+	ExtensionConfiguration *envoy_extensions_filters_http_wasm_v3.Wasm
+}
+
+func convertToWasmPluginWrapper(plugin *config.Config) *WasmPluginWrapper {
+	if wasmPlugin, ok := plugin.Spec.(*extensions.WasmPlugin); ok {
+		cfg := &any.Any{}
```

Comment:
This feels a bit awkward to me, although it may be correct. We take a Struct, covert it to a json string, then create a StringValue then wrap it in Any?

In the API discussion we agreed on JSON but I expected this to be:  We take a Struct, then wrap it in Any. Does that not end up the same/equivalent inside the WASM VM?

Just not sure the difference between the two on the VM side. If the struct ends up in the wrong format then I guess this is the way to go

Reply:
Yeah tbh I was hoping it would have been easier. I spent hours just on this part. The docs say if you put in a Struct it outputs JSON, but doing that didn't work, at least with the extension I tested with.. This was the only solution I got working. There probably is a simpler/better way though